### PR TITLE
Setup server development environment

### PR DESCRIPTION
**Pull request by karenfrilya97**
### What is in this PR:
Preparing the development environment for the backend, except for inserting boilerplate express code
- Adding nodemon for development automation
- Setting up of Google TypeScript Style (gts) to enforce style guide
- Adding README.md to give instructions to develop this application

### CODE REVIEW

**Comment by karenfrilya97** [comment]
> 😱 You missed out your `.gitignore` file, all the node_modules/ are in the PR

Oops D: Thanks for noticing that!

File: server/package.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
+{
+  "name": "gpay-group-buy-server",
+  "version": "1.0.0",
+  "main": "index.js",
+  "repository": "git@github.com:googleinterns/gpay-group-buy.git",
+  "author": "Karen Frilya Celine <karenfrilya@google.com>",
+  "license": "MIT",
```
Comment (hzxcaryn) [inline server/package.json:7]: I think we are using "Apache-2.0" not "MIT" license!

Comment (karenfrilya97) [reply server/package.json:7]: Right! Thanks for noticing :)

File: server/README.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+In the project directory, you can run:
+
+### `yarn`
+This will install all of the dependencies you need.
+
+### `yarn start`
+
+Runs the app in the development mode.<br />
+Open [http://localhost:5000](http://localhost:5000) to view it in the browser.
```
Comment (karenfrilya97) [inline server/README.md:18]: Will add about postman here

File: server/README.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+The page will reload if you make edits.<br />
+You will also see any lint errors in the console.
+
+### `yarn lint`
```
Comment (karenfrilya97) [inline server/README.md:23]: will add that this is about gts as commented by yifu in https://github.com/googleinterns/gpay-group-buy/pull/21#discussion_r439259564

File: server/README.md [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+Receives REST API requests at [http://localhost:5000](http://localhost:5000).
+You can use [Postman](https://www.postman.com/downloads/) to send REST requests
+and view the responses.<br />
+
+The page will reload if you make edits.
+
+### `yarn lint`
+
+Lints and check for formatting problems according to [Google Typescript Style]
+(https://github.com/google/gts).
```
Comment (ifhuang) [inline server/README.md:27]: shouldn't have new line here. (check markdown preview)

File: server/README.md [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+### `yarn fix`
+
+Automatically fixes formatting and linting problems (if possible) according to
+[Google Typescript Style] (https://github.com/google/gts).
```
Comment (ifhuang) [inline server/README.md:32]: remove the space after ]

File: server/package.json [hunk 6]
```diff
@@ -0,0 +1,27 @@
+{
+  "name": "gpay-group-buy-server",
+  "version": "1.0.0",
+  "main": "src/index.js",
+  "repository": "git@github.com:googleinterns/gpay-group-buy.git",
+  "author": "Karen Frilya Celine <karenfrilya@google.com>",
```
Comment (ifhuang) [inline server/package.json:6]: probably remove your email, otherwise your email might be crawled and spammed

**Pull request merged**